File: flink-app/src/main/java/com/example/ctr/infrastructure/flink/sink/RedisSink.java
package com.example.ctr.infrastructure.flink.sink;

import com.example.ctr.config.RedisProperties;
import com.example.ctr.domain.model.CTRResult;
import com.example.ctr.infrastructure.flink.sink.impl.RedisRichSink;
import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;

public class RedisSink {

    private final RedisProperties redisProperties;

    public RedisSink(RedisProperties redisProperties) {
        this.redisProperties = redisProperties;
    }

    public RichSinkFunction<CTRResult> createSink() {
        return new RedisRichSink(redisProperties.getHost(), redisProperties.getPort());
    }
}

File: flink-app/src/main/kotlin/com/example/ctr/infrastructure/flink/sink/RedisSink.kt
package com.example.ctr.infrastructure.flink.sink

import com.example.ctr.config.RedisProperties
import com.example.ctr.domain.model.CTRResult
import com.example.ctr.infrastructure.flink.sink.impl.RedisRichSink
import org.apache.flink.streaming.api.functions.sink.RichSinkFunction

class RedisSink(private val redisProperties: RedisProperties) {

    fun createSink(): RichSinkFunction<CTRResult> = RedisRichSink(redisProperties.host, redisProperties.port)
}
